rooms/views.py

```diff
@@ -0,0 +1,114 @@
+from django.shortcuts import render, redirect, reverse, get_object_or_404
+from django.views.decorators.http import require_http_methods
+from django.views.decorators.cache import cache_page
+from django.views.decorators.vary import vary_on_cookie
+from django.http import JsonResponse
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.forms.models import model_to_dict
+from django.contrib import messages
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from conferences.models import Zosia, UserPreferences
+from .models import Room, UserRoom
+
+
+# Cache hard (15mins)
+@vary_on_cookie
+@cache_page(60 * 15)
+@login_required
+@require_http_methods(['GET'])
+def index(request):
+    # Return HTML w/ rooms layout
+    zosia = Zosia.objects.find_active()
+    if not zosia:
+        messages.error(request, _('There is no active conference'))
+        return redirect(reverse('index'))
+
+    preferences = UserPreferences.objects.filter(zosia=zosia, user=request.user).first()
```

Comment:
I guess I think this way is simpler - no need to know what django throws when model is not found - and how error handling affects variable declarations, etc.